src/herder/TransactionQueue.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "crypto/SecretKey.h"
+#include "herder/TxSetFrame.h"
+#include "transactions/TransactionFrame.h"
+#include "util/HashOfHash.h"
+#include "util/XDROperators.h"
+#include "xdr/Stellar-transaction.h"
+
+#include <deque>
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+namespace stellar
+{
+
+class Application;
+
+struct AccountTransactionsState
+{
+    SequenceNumber mMaxSeq{0};
+    int64_t mTotalFees{0};
+
+    friend bool operator==(AccountTransactionsState const& x,
+                           AccountTransactionsState const& y);
+};
+
+enum class TransactionSubmitStatus
+{
+    PENDING = 0,
+    DUPLICATE,
+    ERROR,
```

Comment:
those names are too generic and conflict with other defines (for example, `ERROR` is defined as `0` on my machine)

Reply:
You can just name them `STATUS_PENDING`, `STATUS_DUPLICATE`, etc